quactography/solver/qaoa_solver.py

```diff
@@ -0,0 +1,84 @@
+from qiskit.primitives import Estimator, Sampler
+from qiskit.circuit.library import QAOAAnsatz
+from scipy.optimize import minimize
+import numpy as np
+
+from quactography.solver.io import save_optimization_results
+
+
+# Function to find the shortest path in a graph using QAOA algorithm with parallel processing:
+def _find_longest_path(args):
```

Comment:
Ususally functions that start with an underscore are only used in the same file. There is no other function in this file. I would suggest you add a function `find_longest_path` (without the underscore at the beginning) which takes `hamiltonians, reps, nbr_processes, output_file` as input and simply always run the content of your `multiprocess_qaoa_solver` function. So remove the `multiprocess_qaoa_solver` function (and thus the `multiprocess_solver.py` file) and add its content in a function called `find_longest_path` here. If you have `nbr_processes == 1`, the `multiprocessing.Pool` function will only use one processor. This comment should make sense with further comments from `find_max_intensity_diffusion_path.py`